#165436587 Integrated database to my challenge2

Wokoro opened the pull request:
#### What does this PR do?
Integrate Postgres database with my challenge-2 endpoints

#### Description of Task to be completed?
This is an implementation of part of challenge-3, All endpoints from challenge-2 are now implemented to work with the Postgres database.
###### current endpoints
1. User signup POST auth/signup
2. User signin POST auth/signin
3. Get a single account GET accounts/<account-number>
4. Get all accounts GET accounts/
5. Create account POST accounts/
6. Delete account DELETE accounts/
7. Activate/Deactivate account PATCH account<account-number>
8. Get all transactions GET transactions/
9. Get a single transaction GET transaction/<transactions-number>
10. Credit a user bank account POST transaction/<account-number>/credit
11. Debit a user bank account POST transaction/<account-number>/debit


#### How should this be manually tested?
none

#### Any background context you want to provide?
The repository can be cloned and tested with postman, Heroku currently not enabled.

#### What are the relevant pivotal tracker stories?
[#165436587](https://www.pivotaltracker.com/story/show/165436587)

#### Screenshots (if appropriate)
none

#### Questions:
none

Wokoro committed:
Integrated database to my challenge2

    Swiched endpoints to start using postgres database

Wokoro committed:
<bug>travis-build-fail

Wokoro committed:
<bug><implemented-feedback>

    Implemented the feedbacks as specified.
    The current implementation does the following
    1. Fixes travis build fail errors
    2. Removed .env file

oladimillion commented on .env:
```diff
@@ -1,4 +1,4 @@
 PUB_KEY = '-----BEGIN PUBLIC KEY-----\nMFswDQYJKoZIhvcNAQEBBQADSgAwRwJAXvAed31PLxc6rb5LOaP9CW14reZUG8C5\n0WnpvDPi/xGeYq0kRuiYIOUQ1PRKPC+q21gWuOsFz/qaSweddsLsAQIDAQAB\n-----END PUBLIC KEY-----'
-PRI_KEY = '-----BEGIN RSA PRIVATE KEY-----\nMIIBOgIBAAJAXvAed31PLxc6rb5LOaP9CW14reZUG8C50WnpvDPi/xGeYq0kRuiY\nIOUQ1PRKPC+q21gWuOsFz/qaSweddsLsAQIDAQABAkBcFHC4XCWM+u7CzSqRAWjO\n2AqQQDGbqVWQszUsgPbNhM6IDV2wczZ0bIlehV0ha4hxHBvD97+YP0x1hsl14u7x\nAiEAobYKAPry4oFRTHge23FrHBSp/IeQ4XzjR1rrdfOlVjMCIQCWSyAYG/cZbLkJ\nq6nwGALpOtAjgQArs+5nAeHPFUL4+wIgOffsrB9JVCLxRs1EmnuU3tMVgH4EVKCV\nRK31/ClkGnsCIQCM9bsNQbVZM ...
\ No newline at end of file
+PRI_KEY = '-----BEGIN RSA PRIVATE KEY-----\nMIIBOgIBAAJAXvAed31PLxc6rb5LOaP9CW14reZUG8C50WnpvDPi/xGeYq0kRuiY\nIOUQ1PRKPC+q21gWuOsFz/qaSweddsLsAQIDAQABAkBcFHC4XCWM+u7CzSqRAWjO\n2AqQQDGbqVWQszUsgPbNhM6IDV2wczZ0bIlehV0ha4hxHBvD97+YP0x1hsl14u7x\nAiEAobYKAPry4oFRTHge23FrHBSp/IeQ4XzjR1rrdfOlVjMCIQCWSyAYG/cZbLkJ\nq6nwGALpOtAjgQArs+5nAeHPFUL4+wIgOffsrB9JVCLxRs1EmnuU3tMVgH4EVKCV\nRK31/ClkGnsCIQCM9bsNQbVZM ...
```
add your `.env` file to the gitignore

Wokoro committed:
<refactor><implementation-of-feedback>: Refactored codebase

    Refactored the codebase to reflect feedback given.
    The current implementation covers.
    1. Using descriptive names for functions and variables
    2. Use of arrow functions

oladimillion commented on database/db.js:
```diff
@@ -0,0 +1,74 @@
+/* eslint-disable no-console */
+import pg from 'pg';
+
+const config = {
```
all DB credentials should exist only on your `.env` file

oladimillion commented on middleware/signin_authentication.js:
```diff
@@ -7,17 +8,21 @@ import UserModel from '../src/models/user.model';
  * @param {string} next
  */
-export default function (req, res, next) {
+export default async function (req, res, next) {
```
Use arrow function

Wokoro replied:
Noted

oladimillion commented on middleware/transaction_validation.js:
```diff
@@ -7,15 +8,17 @@ import transactionModel from '../src/models/transaction.model';
 
 
-function transactionNumberValidation(req, res, next) {
+async function transactionNumberValidation(req, res, next) {
```
Use arrow function

Wokoro replied:
Noted.

oladimillion commented on src/controllers/account.controller.js:
```diff
@@ -13,12 +14,17 @@ export default {
-    if (accounts) { return res.json({ message: 'Operation successful', status: 200, accounts }); }
-
-    return res.json({ status: 400, message: 'No accounts created' });
+  async index(req, res) {
```
Why `index`, give the function a descriptive name

Wokoro replied:
Noted.

oladimillion commented on src/controllers/account.controller.js:
```diff
@@ -27,13 +33,14 @@ export default {
 * @returns {object} returns the transaction details if succesful
 */
-  show(req, res) {
+  async show(req, res) {
```
rename this function, maybe "getDetail" or whatever

Wokoro replied:
Noted

oladimillion commented on src/controllers/transaction.controller.js:
```diff
@@ -7,12 +7,14 @@ export default {
-    if (transactions) { return res.json({ status: 200, transactions }); }
-
-    return res.json({ status: 400, message: 'No transation available' });
+  async index(req, res) {
```
rename this function, `index` is not descriptive enough

Wokoro replied:
Noted.

oladimillion commented on src/controllers/transaction.controller.js:
```diff
@@ -21,34 +23,36 @@ export default {
 * @returns {object} returns the transaction details if succesful
 */
-  show(req, res) {
+  async show(req, res) {
```
give this function a more descriptive name

Wokoro replied:
Noted.

oladimillion commented on src/models/account.model.js:
```diff
@@ -32,85 +35,108 @@ class Account {
-  */
-  static save(account) {
-    Account.store.add(account);
+  static async changeStatus(accountNumber) {
```
Why `static`. Remove it

oladimillion commented on src/models/account.model.js:
```diff
@@ -32,85 +35,108 @@ class Account {
   */
-  static delete(account) {
-    Account.store.delete(account);
+  static async delete(accountnumber) {
```
Why `static`. Remove it

oladimillion commented on src/models/account.model.js:
```diff
@@ -32,85 +35,108 @@ class Account {
   */
-  static findByAccountNumber(accountNumber) {
-    return Account.all().find(account => account.accountNumber === parseInt(accountNumber));
+  static async find(column, value) {
```
Why `static`. Remove it

oladimillion commented on src/models/account.model.js:
```diff
@@ -32,85 +35,108 @@ class Account {
   */
-  static all() {
-    return [...Account.store];
+  static async all() {
```
rename this function to a more descriptive name, maybe `getAllAccountNumber`

Wokoro replied:
Noted.

oladimillion commented on src/models/account.model.js:
```diff
@@ -32,85 +35,108 @@ class Account {
   * Function to generete random account number
   * @returns {Integer}
   */
   static genAccountNumber() {
```
Why `static`. Remove it

oladimillion commented on src/models/transaction.model.js:
```diff
@@ -16,35 +18,37 @@ class Transaction {
-  static findByTransactionID(id) {
-    return Transaction.all().find(transaction => transaction.id === parseInt(id));
-  }
+  static async create(type, amount, cashierID, oldBalance, newBalance, accountNumber) {
```
Why `static`. Remove it

oladimillion commented on src/models/transaction.model.js:
```diff
@@ -16,35 +18,37 @@ class Transaction {
   */
-  static findById(id) {
-    return Transaction.store.find(transaction => transaction.id === id);
+  static async find(column, value) {
```
Why `static`. Remove it

oladimillion commented on src/models/transaction.model.js:
```diff
@@ -16,35 +18,37 @@ class Transaction {
   */
-  static all() {
-    return Transaction.store;
+  static async all() {
```
give the function a more descriptive

Wokoro replied:
Done that.

oladimillion commented on src/models/user.model.js:
```diff
@@ -31,32 +31,41 @@ class User {
   */
-  static findByEmail(email) {
-    return User.all().find(user => user.email === email);
+  static async find(column, value) {
```
Why `static`. Remove it

oladimillion commented on middleware/transaction_view_authentication.js:
```diff
@@ -0,0 +1,31 @@
+import UserModel from '../src/models/user.model';
+import TransactionModel from '../src/models/transaction.model';
+
+export default async (req, res, next) => {
```
Give the function a descriptive name

oladimillion commented on src/models/user.model.js:
```diff
@@ -31,32 +31,41 @@ class User {
   */
-  static all() {
-    return [...User.store];
+  static async all() {
```
Give the function a more descriptive name

Wokoro replied:
Ok. noted
